modules/zip/src/parse-zip/zip-compozition.ts

```diff
@@ -111,3 +116,74 @@ export async function addOneFile(zipUrl: string, fileToAdd: ArrayBuffer, fileNam
     )
   );
 }
+
+/**
+ * creates zip archive with no compression
+ * @param inputPath path where files for the achive are stored
+ * @param outputPath path where zip archive will be placed
+ */
+export async function createZip(inputPath: string, outputPath: string) {
+  const fileIterator = await getFileIterator(inputPath);
+
+  const resFile = new NodeFile(outputPath, 'w');
+
+  const cdArray: ArrayBuffer[] = [];
+  for await (const file of fileIterator) {
+    const size = (await resFile.stat()).bigsize;
+    const [localPart, cdHeaderPart] = await generateFileHeaders(file.path, file.file, size);
+    await resFile.append(localPart);
+    cdArray.push(cdHeaderPart);
+  }
+  // TODO add hash file there
+  const cdOffset = (await resFile.stat()).bigsize;
+  const cd = concatenateArrayBuffers(...cdArray);
+  await resFile.append(new Uint8Array(cd));
+  const eoCDStart = (await resFile.stat()).bigsize;
+  await resFile.append(
+    new Uint8Array(
+      generateEoCD({recordsNumber: cdArray.length, cdSize: cd.byteLength, cdOffset, eoCDStart})
+    )
+  );
+}
+
+/**
+ * creates iterator providing buffer with file content and path to every file in the input folder
+ * @param inputPath path to the input folder
+ * @returns iterator
+ */
+export async function getFileIterator(
+  inputPath: string
+): Promise<AsyncIterable<{path: string; file: ArrayBuffer}>> {
+  const fileList = await getAllFiles(inputPath);
+  const iterable = (async function* () {
```

Comment:
```suggestion
  const iterable = (async function* () {
    const fileList = await getAllFiles(inputPath);
```
If you put this inside the asynciterable, maybe the other function does not need to be async? The async iterable is already async≥

Reply:
Corrected